### PR TITLE
Update readme and add new training materials

### PR DESCRIPTION
**Pull request by stichbury**
The new set of training pages sits in a separate folder (for now) called `training_docs` 

This is one PR where it's probably easier to read the files through in preview than to attempt any diffs.

### CODE REVIEW

File: training_docs/13_custom_datasets.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Custom datasets
+
+Kedro supports a number of datasets out of the box, but you can also add support for any proprietary data format or filesystem in your pipeline.
+
+You can find further information about [how to add support for custom datasets](https://kedro.readthedocs.io/en/stable/07_extend_kedro/03_custom_datasets.html) in specific documentation covering advanced usage.
+
+## Contributing a custom dataset implementation
+
+One of the easiest ways to contribute back to Kedro is to share a custom dataset. Kedro has a :code:`kedro.extras.datasets` sub-package where you can add a new custom dataset implementation to share it with others. You can find out more in the [Kedro contribution guide](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) on Github.
```
Comment (MerelTheisenQB) [inline training_docs/13_custom_datasets.md:9]: I'm not sure what `:code:` should do here?

**Review by MerelTheisenQB — APPROVED** [review]
This looks good to me! 👍  Just some minor comments from my side.

File: training_docs/03_new_project.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+kedro new
+```
+
+Feel free to name your project as you like, but this guide will assume the project is named **`Kedro Training`**.
+
+Keep the default names for the `repo_name` and `python_package` when prompted.
+
+
+### Project structure
+Take a few minutes to familiarise yourself with the [Kedro project structure](https://kedro.readthedocs.io/en/stable/02_get_started/05_example_project.html#project-directory-structure) by exploring the contents of `kedro-training` folder.
```
[the remark this answers is not in the document]
Comment (stichbury) [reply training_docs/03_new_project.md:69]: This means the kedro-training folder that you've created when you did the `kedro new` and called the project Kedro Training. I shall clarify.

File: training_docs/06_jupyter_notebook_workflow.md [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+![](./images/jupyter_create_new_notebook.png)
+
+> *Note:* The only kernel available by default has a name of the current project. If you need to access all available kernels, add `--all-kernels` to the command above.
+
+Every time you start or restart a Jupyter or IPython session in the CLI using a `kedro` command, a startup script in `.ipython/profile_default/startup/00-kedro-init.py` is executed. It adds the following variables in scope:
+
+* `context` (`KedroContext`) - Kedro project context that provides access to Kedro's library components.
+* `catalog` (`DataCatalog`) - Data catalog instance that contains all defined datasets; this is a shortcut for `context.catalog`
+* `startup_error` (`Exception`) - An error that was raised during the execution of the startup script or `None` if no errors occurred
```
[the remark this answers is not in the document]
Comment (stichbury) [reply training_docs/06_jupyter_notebook_workflow.md:78]: This was a straight copy from the docs, so will need an update in both places.

File: training_docs/13_custom_datasets.md (hunk 1 above)
Comment (stichbury) [reply training_docs/13_custom_datasets.md:9]: Me neither 😆 
This is another straight copy from the docs. I shall remove in both!

File: training_docs/06_jupyter_notebook_workflow.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (stichbury) [reply training_docs/06_jupyter_notebook_workflow.md:79]: @MerelTheisenQB I added a sentence in [this commit](https://github.com/quantumblacklabs/kedro-training/pull/15/commits/aff347801a5f80115ff2eefbea8f0432edc2345e) 
Please could you take a look and confirm that is OK? Feel free to change it!

[the remark this answers is not in the document]
Comment (MerelTheisenQB) [reply training_docs/06_jupyter_notebook_workflow.md:79]: This looks good to me 🙂

File: training_docs/01_welcome.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+* [Configuration](./11_configuration.md)
+* [Transcoding](./12_transcoding.md)
+* [Custom datasets](./13_custom_datasets.md)
+* [Custom CLI commands](./14_custom_cli_commands.md)
+* [Kedro plugins](./15_plugins.md)
+
+
+## Before your training session
+
+These training materials assume some level of technical understanding.
```
Comment (yetudada) [inline training_docs/01_welcome.md:24]: I'd ask "What does technical mean?"

File: training_docs/02_prerequisites.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Kedro supports [Python 3.6, 3.7 or 3.8](https://www.python.org/downloads/) so you should make sure that it is installed on your laptop. 
+
+We recommend using [Anaconda](https://www.anaconda.com/download) (Python 3.7 version) to install Python packages. You can also use `conda`, which comes with Anaconda, as a virtual environment manager when you install Kedro.
+
+## Kedro
+Follow the official [Kedro prerequisites documentation](https://kedro.readthedocs.io/en/latest/02_get_started/01_prerequisites.html) and the page that follows it to [install Kedro](https://kedro.readthedocs.io/en/latest/02_get_started/02_install.html).
+
+A clean install of Kedro is designed to be lightweight. If you later need additional tools, it is possible to install them later.
+
+>**Note**: If you encounter any problems, please engage Kedro community support on [Stack Overflow](https://stackoverflow.com/questions/tagged/kedro).
```
Comment (yetudada) [inline training_docs/02_prerequisites.md:23]: Can we mention Discourse here too?

**Review by limdauto — APPROVED** [review]

**Review by yetudada — APPROVED** [review]
Minor comments but I'm otherwise really happy! Thank you @stichbury 🥇

File: training_docs/01_welcome.md (hunk 4 above)
Comment (stichbury) [reply training_docs/01_welcome.md:24]: Good catch. Fixed!

File: training_docs/02_prerequisites.md (hunk 5 above)
Comment (stichbury) [reply training_docs/02_prerequisites.md:23]: Added.

**Pull request merged**